### PR TITLE
Reset and retry rebase upon PR regeneration conflicts

### PR DESCRIPTION
**Pull request by brandong954**
This PR adds back the mistakenly removed [retry logic](https://github.com/brandong954/jx3-pipeline-catalog/commit/eb1c3fbf622729d0f7a23ebceeff0a7d203a7f63) within #1470 since e previous rebase attempt can change the initial state of the PR's branch, and thus cause the `regeneration` commit to see conflicts when picked in a successive rebase attempt. Note, this fix is exclusive to itself and not related to the recent [fix](https://github.com/jenkins-x/jx3-pipeline-catalog/pull/1470) for ensuring all `regeneration` commits **outside** of the PR's branch are rebased without conflict.

In other words, the `-r` flag fixed rebase conflicts for `regeneration` commits **outside** of the PR’s branch, but not within it, so both pieces are needed to hopefully resolve all the rebase issues.

### CODE REVIEW

**Comment by msvticket** [comment]
What problem are you trying to solve with this PR?

**Comment by brandong954** [comment]
I've updated the description of the PR to better highlight the problem, but please do let me know if there's anything else I can help clarify.

**Comment by brandong954** [comment]
@msvticket , what's the best way to get these changes reflected within[ jx3-versions](https://github.com/jenkins-x/jx3-versions/blob/master/git/github.com/jenkins-x/jx3-pipeline-catalog.yml#L2)?

**Comment by msvticket** [comment]
If a new version of jx is release so is jx3-pipeline-catalog. I have some PR I'll merge to get the automated process started...
